### ISSUE TITLE
Simplify endian-specific code

### ISSUE THREAD

**Issue by briansmith** [briansmith/ring#94]
This is a common pattern throughout the code, especially in crypto/modes/gcm.c:
```
    const union {
      long one;
      char little;
    } is_endian = {1};
```
...
```
    if (is_endian.little) {
#ifdef BSWAP8
      Xi[0] = BSWAP8(Z.hi);
      Xi[1] = BSWAP8(Z.lo);
#else
      uint8_t *p = (uint8_t *)Xi;
      uint32_t v;
      v = (uint32_t)(Z.hi >> 32);
      PUTU32(p, v);
      v = (uint32_t)(Z.hi);
      PUTU32(p + 4, v);
      v = (uint32_t)(Z.lo >> 32);
      PUTU32(p + 8, v);
      v = (uint32_t)(Z.lo);
      PUTU32(p + 12, v);
#endif
    } else {
      Xi[0] = Z.hi;
      Xi[1] = Z.lo;
    }
```

AFAICT, code sequences like this can be replaced with something much simpler like:
```
      Xi[0] = from_be_u64(Z.hi);
      Xi[1] = from_be_u64(Z.lo);
```
or:
```
      Xi[0] = from_be_u64_ptr(&Z.hi);
      Xi[1] = from_be_u64_ptr(&Z.lo);
```

We should make this so. There's already work underway to replace `BSWAP4`/`BSWAP8`/`GETU32`/`PUTU32`/etc. with `from_be_u32` and friends, and this should be done on top of that.

**Comment by briansmith** [comment]
The tricky part of this work is that it needs to be measured on 32-bit and 64-bit platforms using GCC 5+ and clang 3.7+ and MSVC 2015. I will do the MSVC 2015 measurements. In particular, the relative performance of the `_ptr` form vs non-`_ptr` forms of the functions need to be investigated for the cases where either one would work.

**Comment by briansmith** [comment]
This was partially fixed in #91 with c882c2c3ec837f14267991122dc20e6aba5700bc. However, the annoying code pattern mentioned in the initial comment of this issue still remains.

**Comment by briansmith** [comment]
AFAICT, this is done as of 7d5fbab93089a5183f5e34c33d0cca0c81634505.
